### PR TITLE
Update Object Headers with latest findings

### PR DESCRIPTION
**Pull request by LeftofZen**
A lot of miscellaneous fixes/corrections I've found in my travels through openloco recently. Not ready to merge yet, I need to clean up the branch/PR

### CODE REVIEW

File: src/OpenLoco/src/Objects/IndustryObject.h [hunk 1]
```diff
@@ -85,20 +85,20 @@ namespace OpenLoco
...
-        const uint8_t* buildingParts[32];                    // 0x3C Access with getBuildingParts helper method
-        uint8_t minNumBuildings;                             // 0xBC
-        uint8_t maxNumBuildings;                             // 0xBD
-        const uint8_t* buildings;                            // 0xBE
-        uint32_t availableColours;                           // 0xC2 bitset
-        uint32_t buildingSizeFlags;                          // 0xC6 flags indicating the building types size 1:large4x4, 0:small1x1
-        uint16_t designedYear;                               // 0xCA start year
-        uint16_t obsoleteYear;                               // 0xCC end year
+        uint8_t buildingVariationCount;
+        uint8_t buildingVariationCount2;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Objects/IndustryObject.h:89]: this is the parts count

File: src/OpenLoco/src/Objects/IndustryObject.h [hunk 2]
```diff
@@ -85,20 +85,20 @@ namespace OpenLoco
...
+        const uint8_t* buildingVariationHeights;                  // 0x20 This is the height of a building image
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Objects/IndustryObject.h:90]: I called these buildingPartHeights as you could in theory build multiple building variations from the same set of parts. If anything i think 0x3C should be buildingVariations and these (0x20, 0x24) should be all parts.

File: src/OpenLoco/src/Objects/IndustryObject.h [hunk 3]
```diff
@@ -85,20 +85,20 @@ namespace OpenLoco
...
-        const IndustryObjectUnk38* var_38;                   // 0x38 Access with getUnk38 helper method
-        const uint8_t* buildingParts[32];                    // 0x3C Access with getBuildingParts helper method
-        uint8_t minNumBuildings;                             // 0xBC
-        uint8_t maxNumBuildings;                             // 0xBD
-        const uint8_t* buildings;                            // 0xBE
-        uint32_t availableColours;                           // 0xC2 bitset
-        uint32_t buildingSizeFlags;                          // 0xC6 flags indicating the building types size 1:large4x4, 0:small1x1
-        uint16_t designedYear;                               // 0xCA start year
-        uint16_t obsoleteYear;                               // 0xCC end year
+        uint8_t buildingVariationCount;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Objects/IndustryObject.h:88]: this is the opposite of 0x1F. One is the variations one is the parts.

Comment (duncanspumpkin) [inline src/OpenLoco/src/Objects/IndustryObject.h:88]: Also could you add the offsets if you are naming them 0x1E and 0x1F.

File: src/OpenLoco/src/Objects/BuildingObject.h [hunk 4]
```diff
@@ -28,36 +28,45 @@ namespace OpenLoco
...
-        uint8_t var_A4[2];                  // 0xA4 Some type of Cargo
-        uint8_t var_A6[2];                  // 0xA6
-        uint8_t var_A8[2];                  // 0xA8
-        int16_t demolishRatingReduction;    // 0XAA
-        uint8_t var_AC;                     // 0xAC
-        uint8_t var_AD;                     // 0XAD
-        const uint8_t* var_AE[4];           // 0XAE ->0XB2->0XB6->0XBA->0XBE (4 byte pointers)
+        StringId name;                                     // 0x0
+        uint32_t image;                                    // 0x2
+        uint8_t numAnimations;                             // 0x6
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Objects/BuildingObject.h:46]: this is the number of parts (see industry object)

**Comment by duncanspumpkin** [comment]
The naming of the building parts/variations has always been a bit rubbish but i think we should go for the following:

A building variation is a full displayed different building.
A building part is the individual parts that make up a variation. 

So lets see how that works:
Here we have the steel mill building variation 0. It is comprised of two parts base and body.
image

Here is variation 4. It is again comprised of two parts base and body.
image

In theory a variations can share parts (although in this steel mill example they do not).

File: src/OpenLoco/src/Objects/BuildingObject.h [hunk 5]
```diff
@@ -28,36 +28,45 @@ namespace OpenLoco
     OPENLOCO_ENABLE_ENUM_OPERATORS(BuildingObjectFlags);
 
 #pragma pack(push, 1)
+
+    // copied from IndustryObject.h, these should share a common definition
+    struct BBuildingPartAnimation
```
[the remark this answers is not in the document]
Comment (LeftofZen) [reply src/OpenLoco/src/Objects/BuildingObject.h:33]: no reason, other than its an identical struct to `BuildingPartAnimation` in IndustryObject.h, and this was my placeholder as I haven't figured out where to put common structs yet. This duplicate struct will be deleted in my next change and both Industry and Building will use the same struct

File: src/OpenLoco/src/Objects/DockObject.cpp [hunk 6]
```diff
@@ -60,7 +60,7 @@ namespace OpenLoco
         // Load unk2?
         for (auto i = 0U; i < numAux02Ent; ++i)
         {
-            var_1C[0] = reinterpret_cast<const uint8_t*>(remainingData.data());
+            var_1C[i] = reinterpret_cast<const uint8_t*>(remainingData.data());
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Objects/DockObject.cpp:63]: oh lol

**Review by duncanspumpkin — APPROVED** [review]
I've addressed the comments i made and Aaron made. Any further misc fixes can go in another PR.

**Comment by AaronVanGeffen** [comment]
Changes look good to me. Could we change the PR title to something more descriptive before we (squash) merge this, though?

**Pull request merged**